bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/HandlerPercentageController.java

```diff
@@ -0,0 +1,143 @@
+                int value = state.get("value").getAsInt();
+                // For groups take the maximum
+                if (percentageValue == null) {
+                    percentageValue = value;
+                } else if (value > percentageValue) {
+                    percentageValue = value;
+                }
+            }
+        }
+        if (percentageValue != null) {
+            lastPercentage = percentageValue;
+        }
+        updateState(PERCENTAGE.channelId, percentageValue == null ? UnDefType.UNDEF : new PercentType(percentageValue));
+    }
+
+    @Override
+    public boolean handleCommand(Connection connection, SmartHomeDevice shd, String entityId,
+            SmartHomeCapability[] capabilties, String channelId, Command command) throws IOException {
+        if (channelId.equals(PERCENTAGE.channelId)) {
+            if (containsCapabilityProperty(capabilties, PERCENTAGE.propertyName)) {
+                if (command.equals(IncreaseDecreaseType.INCREASE)) {
+                    Integer lastPercentage = this.lastPercentage;
+                    if (lastPercentage != null) {
+                        int newValue = lastPercentage++;
+                        if (newValue > 100) {
+                            newValue = 100;
+                        }
+                        this.lastPercentage = newValue;
+                        connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName, newValue);
+                        return true;
+                    }
+                } else if (command.equals(IncreaseDecreaseType.DECREASE)) {
+                    Integer lastPercentage = this.lastPercentage;
+                    if (lastPercentage != null) {
+                        int newValue = lastPercentage--;
+                        if (newValue < 0) {
+                            newValue = 0;
+                        }
+                        this.lastPercentage = newValue;
+                        connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName, newValue);
+                        return true;
+                    }
+                } else if (command.equals(OnOffType.OFF)) {
+                    lastPercentage = 0;
+                    connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName, 0);
+                    return true;
+                } else if (command.equals(OnOffType.ON)) {
+                    lastPercentage = 100;
+                    connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName, 100);
+                    return true;
+                } else if (command instanceof PercentType) {
+                    lastPercentage = ((PercentType) command).intValue();
+                    connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName,
+                            ((PercentType) command).floatValue() / 100);
```

Comment:
All of other methods suggest that `setPercentage` is supposed to range from 0-100, not 0-1. Is it supposed to be this instead?
```suggestion
                    connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName,
                            lastPercentage);
```

Reply:
Also changed to int, as states are stored with int.